internal/charmstore/addentity_test.go

```diff
@@ -462,12 +489,23 @@ func assertBlobFields(c *gc.C, doc *mongodoc.Entity, url *router.ResolvedURL, ha
 	if doc.CharmMeta != nil && len(doc.CharmMeta.Series) > 0 {
 		// It's a multi-series charm, so the PreV5* fields should be active.
 		if doc.PreV5BlobSize <= doc.Size {
-			c.Fatalf("pre-v5 blobsize %d is unexpectedly less than original blob size %d", doc.PreV5BlobSize, doc.Size)
+			c.Fatalf("pre-v5 blobsize %d is unexpectedly less than or equal to original blob size %d", doc.PreV5BlobSize, doc.Size)
 		}
 		c.Assert(doc.PreV5BlobHash, gc.Not(gc.Equals), "")
 		c.Assert(doc.PreV5BlobHash, gc.Not(gc.Equals), hash)
 		c.Assert(doc.PreV5BlobHash256, gc.Not(gc.Equals), "")
 		c.Assert(doc.PreV5BlobHash256, gc.Not(gc.Equals), hash256)
+	} else if url.URL.Series == "bundle" && doc.PreV5BlobHash != doc.BlobHash {
+		// It's a bundle with a different PreV5BlobHash, check
+		// that all other fields are consistently different.
+		// TODO(mhilton) find a better way to detemine the kind
```

Comment:
// TODO(mhilton) use BundleData.UnmarshaledWithServices to determine whether to make this check.

?